package.json

```diff
@@ -58,55 +58,54 @@
   },
   "homepage": "https://kulshekhar.github.io/ts-jest",
   "dependencies": {
-    "bs-logger": "^0.2.5",
-    "buffer-from": "^1.1.1",
-    "fast-json-stable-stringify": "^2.0.0",
-    "json5": "^2.0.1",
-    "make-error": "^1.3.5",
-    "mkdirp": "^0.5.1",
-    "semver": "^5.5.1",
-    "yargs-parser": "^10.1.0"
+    "bs-logger": "0.x",
+    "buffer-from": "1.x",
+    "fast-json-stable-stringify": "2.x",
+    "json5": "2.x",
+    "make-error": "1.x",
+    "mkdirp": "0.x",
+    "semver": "5.x",
+    "yargs-parser": "10.x"
   },
   "peerDependencies": {
     "jest": ">=22 <24"
   },
   "devDependencies": {
-    "@commitlint/cli": "^7.1.2",
-    "@commitlint/config-conventional": "^7.1.2",
-    "@types/babel__core": "^7.0.1",
-    "@types/buffer-from": "^1.1.0",
-    "@types/cross-spawn": "^6.0.0",
-    "@types/fs-extra": "5.0.4",
-    "@types/jest": "^23.3.2",
-    "@types/js-yaml": "^3.11.2",
-    "@types/json5": "^0.0.30",
-    "@types/lodash.memoize": "^4.1.4",
-    "@types/lodash.merge": "^4.6.4",
-    "@types/lodash.set": "^4.3.4",
-    "@types/mkdirp": "^0.5.2",
-    "@types/node": "^10.9.4",
-    "@types/semver": "^5.5.0",
-    "@types/yargs": "^11.1.1",
-    "conventional-changelog-cli": "^2.0.5",
-    "cross-spawn": "^6.0.5",
-    "doctoc": "^1.3.1",
-    "eslint": "^5.5.0",
-    "fs-extra": "^7.0.0",
-    "glob-gitignore": "^1.0.9",
-    "husky": "^0.14.3",
-    "jest": "^23.6.0",
-    "js-yaml": "^3.12.0",
-    "lint-staged": "^7.2.2",
-    "lodash.memoize": "^4.1.2",
-    "lodash.merge": "^4.6.1",
-    "lodash.set": "^4.3.2",
-    "npm-run-all": "^4.1.3",
-    "prettier": "^1.14.2",
-    "source-map": "^0.7.3",
-    "tslint": "^5.11.0",
-    "tslint-config-prettier": "^1.15.0",
-    "tslint-plugin-prettier": "^1.3.0",
-    "typescript": "^3.0.3"
+    "@commitlint/cli": "7.x",
+    "@commitlint/config-conventional": "7.x",
+    "@types/babel__core": "7.x",
+    "@types/buffer-from": "*",
+    "@types/cross-spawn": "*",
+    "@types/fs-extra": "*",
+    "@types/jest": "23.x",
+    "@types/js-yaml": "*",
+    "@types/json5": "*",
+    "@types/lodash.memoize": "4.x",
+    "@types/lodash.merge": "4.x",
+    "@types/lodash.set": "4.x",
+    "@types/mkdirp": "*",
+    "@types/node": "10.x",
+    "@types/semver": "*",
+    "@types/yargs": "*",
+    "conventional-changelog-cli": "2.x",
+    "cross-spawn": "*",
+    "eslint": "*",
+    "fs-extra": "*",
```

Comment:
are `*` and `.x` a good idea? It looks very risky and could let breaking changes in. It won't allow greenkeeper to do its job either, right?

Reply:
- `*` are for dev deps only, if it breaks, it's tests on local machines before anything
- `major.x` are for non dev deps which need to be locked to a certain version, or dev deps which must be locked on major version (for changelog for example)

I'll look at how greenkeeper is working, but I thought that it was clearer to read what may come in as versions.

Don't forget that caret notation (`^M.m.p`) is the same as `M.x`.

Reply:
right, I didn't notice this was only for dev deps. This looks fine then :+1: